frame/dvm/ethereum/src/lib.rs

```diff
@@ -309,6 +315,30 @@ impl<T: Trait> Module<T> {
 		frame_system::Module::<T>::deposit_log(digest.into());
 	}
 
+	/// Get the remaining balance for evm address
+	pub fn remaining_balance(account_id: &T::AccountId) -> T::Balance {
+		<RemainingBalance<T>>::get(account_id)
+	}
+
+	/// Inc remaining balance
+	pub fn inc_remain_balance(account_id: &T::AccountId, value: T::Balance) {
+		let remain_balance = Self::remaining_balance(account_id);
+		let updated_balance = remain_balance
+			.checked_add(&value)
+			.expect("Inc remain balance, panic!");
```

Comment:
Shouldn't panic here.

Reply:
Since the remaining balance is the cal result of mod 10^9,  No panic here.

Use expect rather than return error to avoid defining a new type error.

Reply:
`saturating_add` would be better

Reply:
`panic` and returning `error` has different meanings.

> Use expect rather than return error to avoid defining a new type error.

This is not the problem that writing less code or keeping function clean.

### `panic`
We should use `panic` when something went wrong that we can not handle, then, **crash the program**, do we have to crash the program?

### `Error`

Throw errors if it's not expected, then handle it in the top-level if we can.

Reply:
Done